genesis/engine/solvers/base_solver.py

```diff
@@ -34,6 +34,45 @@ def __init__(self, scene: "Scene", sim: "Simulator", options):
     def _add_force_field(self, force_field):
         self._ffs.append(force_field)
 
+    @staticmethod
+    def _is_ti_field(obj) -> bool:
```

Comment:
This is definitely not the right way to check if something is a taichi field. Just do
`isinstance(obj, taichi.Field)` and remove this method entirely. I checked myself and it works perfectly on both `ti.Vector`, `ti.Matrix` and `ti.types.struct` instances.